### ISSUE TITLE
Pull Target Levels from Vanilla Goals for Skill Calculator

### ISSUE THREAD

**Issue by WLoumakis** [runelite/runelite#8879]
This is not related to a problem, per se, but rather a QoL-type issue. The current way the Skill Calculator works is a bit agitating (at least to me). You can click off of a skill and upon clicking back on it the target level gets wiped.

I feel as though it would be much more elegant to utilize the vanilla XP goals, similar to how the XP tracker works, to either extrapolate the target level or the target xp. Then, if the user wishes to see intermediate or post-goal levels, they can change the target level/xp.

Perhaps instead of making this the default, though I don't see why anyone wouldn't want this, an option could be added in Configuration->Skill Calculator to where you can add a clickbox to pull from vanilla goals instead of wiping each time or at least to remember the last input target level/xp. Doesn't have to be in config, but just a thought.

**Comment by WLoumakis** [comment]
Since this hasn't gotten any attention and there has been a similar post (see issue [#10445](https://github.com/runelite/runelite/issues/10445)), I decided to go in and make the changes myself.  The only file changed is runelite-client/src/main/java/net.runelite.client/plugins/skillcalculator/SkillCalculator.java.

The changes can be found on [my forked version](https://github.com/WLoumakis/runelite)

Changes made:
1. Added imports for Skill and VarPlayer.
2. Changed the following two lines of code
`targetLevel = currentLevel + 1;
targetXP = Experience.getLevelForXp(targetLevel);`
to
`VarPlayer endGoal = endGoalVarpForSkill(calculatorType.getSkill());
targetXP = endGoal != null ? client.getVar(endGoal) : -1;
targetLevel = enforceSkillBounds(Experience.getLevelForXp(targetXP));`
3. Added the private static method `endGoalVarpForSkill` method from the XpTrackerPlugin.java file.

One thing to note is that I did not do any testing on this, specifically of concern would be the `targetXP = endGoal ? ...` line since it would potentially set `targetXP` to -1, although I am almost certain that would get caught in the `enforceSkillBounds` call.

All changes are currently live on my fork, but if someone can walk me through creating a pull request here then I can get that done.
